cranelift/codegen/src/isa/aarch64/inst/args.rs

```diff
@@ -220,8 +222,10 @@ impl AMode {
             &AMode::RegExtended(r1, r2, ext) => {
                 AMode::RegExtended(allocs.next(r1), allocs.next(r2), ext)
             }
-            &AMode::PreIndexed(reg, simm9) => AMode::PreIndexed(allocs.next_writable(reg), simm9),
-            &AMode::PostIndexed(reg, simm9) => AMode::PostIndexed(allocs.next_writable(reg), simm9),
+            // Note that SP is not managed by regalloc, so there is no register to report in the
+            // pre/post-indexed amodes.
+            &AMode::SPPreIndexed(simm9) => AMode::SPPreIndexed(simm9),
```

Comment:
Nit - use `self.clone()` as in the match arms that follow?

Reply:
Done!